notify_python35.cpp

```diff
@@ -151,15 +147,12 @@ bool NotifyPython35::configure()
 		return true;
 	}
 
-	// Check the Python interpreter is set
-	if (!Py_IsInitialized())
+	// 2) Import Python script if module object is not set
+	if (!m_pModule)
 	{
-		interpreterStart(this);
+		m_pModule = PyImport_ImportModule(m_pythonScript.c_str());
```

Comment:
I think we should acquire and hold GIL for remainder of this function.

Reply:
NotifyPython35::configure()

is always called while holding the GIL

NotifyPython35::reconfigure
        ....
        PyGILState_STATE state = PyGILState_Ensure();
        ...
        bool ret = this->configure();

        PyGILState_Release(state);


and

PLUGIN_HANDLE plugin_init(ConfigCategory* config)

     PyGILState_STATE state = PyGILState_Ensure(); // acquire GIL
    ....
      // Configure plugin
      notify->lock();
      bool ret = notify->configure();
      notify->unlock();

      PyGILState_Release(state); // release GIL

Reply:
It is better to acquire and release GIL where required. Try not to hold GIL across main functions. Might introduce errors during code changes in future.

Reply:
@AmandeepArora this plugin is now similar to foglamp-filter-python35, same base code for init, configure and reconfigure.

As per the comment we should modify both.

 It's not needed now the change so a new JIRA task could be raised for both plugins